api/children/childrenRouter.js

```diff
@@ -13,185 +19,45 @@ router.get('/', authRequired, function (req, res) {
-        }
-      });
-    } catch (err) {
-      console.error(err);
-      res.status(500).json({ message: err.message });
-    }
-  } else {
-    res.status(404).json({ message: 'Child Details missing.' });
-  }
-});
-
-router.put('/', authRequired, (req, res) => {
-  const child = req.body;
-  if (child) {
-    const { id } = child;
-    Children.findByChildId(id)
-      .then(
-        Children.updateChild(id, child)
-          .then((updated) => {
-            res.status(200).json({
-              message: `Child with id: ${id} updated`,
-              child: updated[0],
-            });
-          })
-          .catch((err) => {
-            res.status(500).json({
-              message: `Could not update child '${id}'`,
-              error: err.message,
-            });
-          })
-      )
-      .catch((err) => {
-        res.status(404).json({
-          message: `Could not find child '${id}'`,
-          error: err.message,
-        });
-      });
+//dont forget authRequired
+router.get('/:id', checkChildExist, authRequired, function (req, res, next) {
+  try {
+    res.status(200).json(req.child);
+  } catch (error) {
+    next(error);
   }
 });
-
-router.delete('/:id', authRequired, (req, res) => {
-  const id = req.params.id;
+//dont forget authRequired
+router.get('/:id/enrollments', checkChildExist, authRequired, async function (
+  req,
+  res,
+  next
+) {
   try {
-    Children.findByChildId(id).then((child) => {
-      Children.removeChild(child[0].id).then(() => {
-        res.status(200).json({
-          message: `Child with id:'${id}' was deleted.`,
-          child: child[0],
-        });
-      });
-    });
-  } catch (err) {
-    res.status(500).json({
-      message: `Could not delete child with ID: ${id}`,
-      error: err.message,
-    });
+    const { id } = req.params;
+    const enrollments = await Children.getEnrolledCourses(id);
+    res.status(200).json(enrollments);
+  } catch (error) {
+    next(error);
   }
 });
-
-router.put('/enrollments', authRequired, (req, res) => {
-  const enrollment = req.body;
-  if (enrollment) {
-    const { id } = enrollment;
-    Children.updateEnrollment(id, enrollment)
-      .then((updated) => {
-        res.status(200).json({
-          message: `Enrollment with id: ${id} updated`,
-          enrollment: updated[0],
-        });
-      })
-      .catch((err) => {
-        res.status(500).json({
-          message: `Could not update enrollment '${id}'`,
-          error: err.message,
-        });
-      })
-
-      .catch((err) => {
-        res.status(404).json({
-          message: `Could not find enrollment '${id}'`,
-          error: err.message,
-        });
-      });
+//dont forget authRequired
+router.post(
+  '/:id/enrollments',
+  checkChildExist,
+  isChildAlreadyEnrolled,
+  authRequired,
+  async (req, res) => {
+    // the req.body is only class_id  === > {class_id} , it will get modified in isChildAlreadyEnrolled middleware
```

Comment:
this is to help the future developer to know what kind of data is expected to be sent to the backend,
this can be added to the documentation later on